### PR TITLE
Add option to specify bind address when connecting

### PR DESCRIPTION
**Pull request by revmischa**
For machines with multiple IPs or people who want to use IPv6

### CODE REVIEW

**Comment by apeiron** [comment]
Can you update this so it merges cleanly, and then we can revisit it?

**Comment by Trinitas** [comment]
#241 possible with Webirc

File: lib/irc.js [hunk 1]
```diff
@@ -573,14 +574,28 @@ Client.prototype.connect = function ( retryCount, callback ) { // {{{
...
+        connectionOpts.localAddress = self.opt.localAddress;
+    
     // try to connect to the server
     if (self.opt.secure) {
         var creds = self.opt.secure;
-        if (typeof self.opt.secure !== 'object') {
-            creds = {};
+        if (typeof self.opt.secure == 'object') {
+            // copy "secure" opts to options passed to connect()
+            for (var f in self.opt.secure) {
```
Comment (jirwin) [inline lib/irc.js]: You'll want to add a `hasOwnProperty` guard here:
```javascript
for (var f in creds) {
  if (creds.hasOwnProperty(f)) {
    connectionOpts[f] = creds[f]
}
```

**Comment by jirwin** [comment]
Looks pretty good to me! add that guard and a rebase and we'll get this merged.

**Comment by revmischa** [comment]
Okay, standby

File: docs/API.rst [hunk 2]
```diff
@@ -39,6 +40,9 @@ Client
     If you set `selfSigned` to true SSL accepts certificates from a non trusted CA.
     If you set `certExpired` to true, the bot connects even if the ssl cert has expired.
 
+    `localAddress` is the address to bind to when connecting. Not
```
Comment (jirwin) [inline docs/API.rst]: It looks like you got this working with the secure option. Probably should update this doc.

**Comment by jirwin** [comment]
Actually, this is based against the wrong branch now. I think you may need to re-open this PR against the `0.3.x` branch

**Comment by revmischa** [comment]
Commit d04df30 makes rebasing highly problematic

**Comment by jirwin** [comment]
Yeah, this is why I think it will probably be easier to just open a new PR against the 0.3.x branch and add the changes there. Rebasing against that will be a huge pain.

**Comment by jirwin** [comment]
I like this PR a lot.

**Comment by apeiron** [comment]
+1. I have a box with a bunch of IPs so I can test this pretty easily. Will look at doing that today and will merge assuming all goes well.

**Pull request merged**